units/event_log_tests.cc

```diff
@@ -0,0 +1,183 @@
+#include <cassert>
+#include <cstddef>
+#include <cstdint>
+#include <cstdio>
+#include <iostream>
+#include <ostream>
+#include <string>
+#include <vector>
+
+#include "gtest/gtest.h"
+
+namespace {
+// Set of supported attribute types.
+enum AttributeType { String, Int64, VectorInt64 };
+// Specifies the importance of an Event. Events are logged based on their level
+// of importance. Eg., compile_time.csv only contains the important compile time
+// events while event.log contains all the compile time events.
+enum LogLevel { Low, Medium, High };
+
+std::string ValueToCSVString(const std::string &value) { return value; }
+
+std::string ValueToCSVString(const int64_t value) {
+  return std::to_string(value);
+}
+
+std::string ValueToCSVString(const std::vector<int64_t> &values) {
+  std::string tmp = "\"[";
+  using std::begin;
+  using std::end;
+  auto e = end(values);
+  for (auto it = begin(values); it != e; ++it) {
+    const char *delimiter = std::next(it) != values.end() ? "," : "";
+    tmp += std::to_string(*it) + delimiter;
+  }
+  tmp += "]\"";
+  return tmp;
+}
+
+struct Attribute {
+  const char *name;
```

Comment:
I think it would be better for `Attribute` to be a `class` with clearly defined public and private interface. For example, we probably don't want all users of `Attribute` to be able to change the name, which we can enforce by making the field `private` and exposing a getter `getName() const;`.